### PR TITLE
Snippets

### PR DESCRIPTION
**Pull request by therealcoder1337**
Same thing as with props, and a few refactors.

### CODE REVIEW

File: docgen/helpers/getComponentDirs.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { readdir } from "fs/promises";
+
+export default async function getComponentDirs(componentsSir: string) {
+  const dirents = await readdir(componentsSir, { withFileTypes: true });
```
Comment (KilDesu) [inline docgen/helpers/getComponentDirs.ts:4]: You probably wanted to write `componentsDir`?

File: docgen/snippets/updateAllSnippets.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+const dirname = path.dirname(fileURLToPath(import.meta.url));
+const rootDir = path.resolve(dirname, "../../src/routes/components");
+
+export default async function updateAllSnippets() {
+  const componentsToIgnore = ["layout"];
+
+  const componentDirs = (await getComponentDirs(rootDir)).filter(
+    (dir) => !componentsToIgnore.includes(dir)
+  );
```
Comment (KilDesu) [inline docgen/snippets/updateAllSnippets.ts:14]:
```js
const componentsDirs = (await getComponentsDirs(rootDir)).filter(!componentsToIgnore.includes)
```

File: src/dev/docgenPlugin.ts [hunk 3]
```diff
@@ -40,12 +47,19 @@ function docgenPlugin(): Plugin {
     name: "docgen-plugin",
     async configResolved(config) {
       config.logger.info(DOCGEN_LOG_MESSAGE);
-      await runDocGenProps();
+      const p1 = runDocGenProps();
+      const p2 = updateAllSnippets();
+      await Promise.all([p1, p2]);
```
Comment (KilDesu) [inline src/dev/docgenPlugin.ts:52]: Do we really need two variables for that, as I don't see them reused somewhere else? Can't we just pass the functions to `Promise.all` directly?

**Review by KilDesu — APPROVED** [review]
I have some comments, but otherwise, lgtm!

File: src/dev/docgenPlugin.ts (hunk 3 above)
Comment (therealcoder1337) [reply src/dev/docgenPlugin.ts:52]: Yeah we could. I find this slightly more readable, but will change it.

**Pull request merged**